openslides/assignments/views.py

```diff
@@ -242,6 +243,134 @@ def get_serializer_class(self):
+            raise ValidationError({'detail': _('Invalid data. Expected dictionary, got %s.') % type(request.data)})
+        user_str = request.data.get('user', '')
+        try:
+            user_pk = int(user_str)
+        except ValueError:
+            raise ValidationError({'detail': _('Invalid data. Expected something like {"user": <id>}.')})
+        try:
+            user = User.objects.get(pk=user_pk)
+        except User.DoesNotExist:
+            raise ValidationError({'detail': _('Invalid data. User %d does not exist.') % user_pk})
+        return user
+
+    @detail_route(methods=['post', 'delete'])
+    def candidature_other(self, request, pk=None):
+        """
+        View to nominate other users (POST) or delete their candidature
+        status (DELETE). The client has to send {'user': <id>}.
+        """
+        if not request.user.has_perm('assignments.can_nominate_other'):
+            self.permission_denied(request)
+        user = self.get_other_user(request)
+        assignment = self.get_object()
+        if assignment.is_elected(user):
+            raise ValidationError({'detail': _('User %s is already elected.') % user})
+        if request.method == 'POST':
+            message = self.nominate_other(request, user, assignment)
+        else:
+            # request.method == 'DELETE'
+            message = self.delete_other(request, user, assignment)
+        return Response({'detail': message})
+
+    def nominate_other(self, request, user, assignment):
+        if assignment.phase == assignment.PHASE_FINISHED:
+            raise ValidationError({'detail': _('You can not nominate someone to this election because it is finished.')})
+        if assignment.phase == assignment.PHASE_VOTING and not request.user.has_perm('assignments.can_manage'):
+            # To nominate other during voting you have to be a manager.
+            self.permission_denied(request)
+        if assignment.is_candidate(user):
+            raise ValidationError({'detail': _('User %s is already a candidate of this election.') % user})
+        if assignment.is_blocked(user) and not request.user.has_perm('assignments.can_manage'):
+            raise ValidationError({'detail': _('User %s does not want to be an candidate.') % user})
+        assignment.set_candidate(user)
+        return _('User %s was nominated successfully.') % user
+
+    def delete_other(self, request, user, assignment):
+        # To delete candidature status you have to be a manager.
+        if not request.user.has_perm('assignments.can_manage'):
+            self.permission_denied(request)
+        if assignment.phase == assignment.PHASE_FINISHED:
+            raise ValidationError({'detail': _('You can not delete someones candidature to this election because it is finished.')})
+        if not assignment.is_candidate(user) and not assignment.is_blocked(user):
+            raise ValidationError({'detail': _('User %s has no status in this election.') % user})
```

Comment:
He can do so. But he can't delete users that are not elected and not candidates or blocked. The concept of this pull request is that we have three different api endpoints where we can send POST or DELETE. The endpoint `candidature_other` is to create new candidates or delete them or for historical reason to delete blocked candidates so that the can be nominated again. The endpoint to mark someone elected or not is something different. If we want to have everything in one endpoint everything else should be not like this. I am not against this way but it would be more difficult to implement.